raster-foundry/raster-foundry
# Enhance project layer permissions for map token sharing and public projects

## Overview

This PR adds map token authorization to listing project layers. Listing project layers for public projects already worked.

### Checklist


### Notes

Prior confusion about analysis sharing:

<details>
It doesn't do analysis map token viewing yet because I'm not sure how that should work. Map tokens can have references to tool runs or to projects. The `listToolRuns` route can in principle be authorized:

- from a map token referencing the tool run
- from a map token referencing the project
- from the tool run being public
- from the project being public
- from the user from the authorization header

Seeking public comment.
</details>

## Testing Instructions

 * reassemble your servers and bring them up + the frontend
 * set a project to shared
 * copy the map token
 * `http :9091/api/projects/<id>/layers mapToken==themaptoken`
 * it should succeed
 * try accessing that project's layers with the token for the owner
 * it should succeed
 * log in to the app with a different user -- it should succeed
 * get that user's token
 * try to access that project's layers with this user's token
 * it should fail
 * log back in as the other user
 * set the project to public
 * send the request for that project's layers without a token
 * it should succeed
 * look at the `projects/id/analyses` route and confirm the logic is the same
 * confirm rendering lab tiles, lab histograms, lab stats, and lab exports still work

Closes #4588
